### ISSUE TITLE
error: exportArchive: exportOptionsPlist error for key 'uploadBitcode': cannot upload bitcode because bitcode is imbalanced

### ISSUE THREAD

**Issue by fzyzcjy** [fastlane/fastlane#17171]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When building, the following error happens 
```
[15:22:17]: $ /usr/bin/xcrun /Library/Ruby/Gems/2.6.0/gems/fastlane-2.158.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/pd/mpc7mwfj36ndyr9xbnnzk_s80000gn/T/gym_config20200906-35526-r2ks4l.plist' -archivePath /Users/tom/Library/Developer/Xcode/Archives/2020-09-06/Runner\ 2020-09-06\ 15.21.22.xcarchive -exportPath '/var/folders/pd/mpc7mwfj36ndyr9xbnnzk_s80000gn/T/gym_output20200906-35526-10phi54' 
+ xcodebuild -exportArchive -exportOptionsPlist /var/folders/pd/mpc7mwfj36ndyr9xbnnzk_s80000gn/T/gym_config20200906-35526-r2ks4l.plist -archivePath '/Users/tom/Library/Developer/Xcode/Archives/2020-09-06/Runner 2020-09-06 15.21.22.xcarchive' -exportPath /var/folders/pd/mpc7mwfj36ndyr9xbnnzk_s80000gn/T/gym_output20200906-35526-10phi54
2020-09-06 15:22:19.428 xcodebuild[39198:11222473] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/pd/mpc7mwfj36ndyr9xbnnzk_s80000gn/T/Runner_2020-09-06_15-22-19.427.xcdistributionlogs'.
error: exportArchive: exportOptionsPlist error for key 'uploadBitcode': cannot upload bitcode because bitcode is imbalanced

Error Domain=IDEFoundationErrorDomain Code=1 "exportOptionsPlist error for key 'uploadBitcode': cannot upload bitcode because bitcode is imbalanced" UserInfo={NSLocalizedDescription=exportOptionsPlist error for key 'uploadBitcode': cannot upload bitcode because bitcode is imbalanced}

...
```

P.S. I see this locked issue, but I am not low on disk (having 180GB free), so that issue does not solve my question https://github.com/fastlane/fastlane/issues/15685

##### Command executed

`fastlane deploy` where my Fastfile is

```
default_platform(:ios)

platform :ios do
  lane :deploy do
    match(type: "appstore")

    Dir.chdir "../.." do
      sh("flutter", "packages", "get")
      sh("flutter", "build", "ios",
         "--release",
         "--no-codesign"
      )
    end

    build_ios_app(
        scheme: "Runner",
        include_bitcode: true,
    )
  end
end
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  too long to be uploaded to github. Please see ---> https://gist.github.com/fzyzcjy/00390e954226c73c9818a750c2f9fa68
</details>

### Environment

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.3                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.3 (19D76)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.7                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | zh_CN.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  lane :deploy do
    # TODO 关于push notification (`get_push_certificate`)
    # https://docs.fastlane.tools/getting-started/ios/appstore-deployment/#best-practices

    # https://docs.fastlane.tools/codesigning/getting-started/#using-cert-and-sigh
    # NOTE Xcode也需要设置 https://docs.fastlane.tools/codesigning/xcode-project/
    match(type: "appstore") # TODO type?

    # https://circleci.com/blog/deploy-flutter-android/
    # 1. "builds against the latest Flutter toolchain"
    Dir.chdir "../.." do
      sh("flutter", "packages", "get")
      # sh("flutter", "clean")
      sh("flutter", "build", "ios",
         # TODO
         # "--obfuscate", "--split-debug-info=build/split-debug-info",
         "--release",
         "--no-codesign"
      )
    end

    # 2. "builds the actual .ipa file that will be uploaded"
    build_ios_app(
        # scheme is seen here https://flutter.dev/docs/deployment/ios#create-a-build-archive
        scheme: "Runner",
        include_bitcode: true,
    )

    # TODO
    # upload_to_testflight / upload_to_app_store / ...
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("com.rzzsdxx.proj") # The bundle identifier of your app
apple_id("ch271828n@outlook.com") # Your Apple email address

itc_team_id("122060207") # App Store Connect Team ID
team_id("QSBDNU2Y9B") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.158.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 4.0.6        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.76.0       |
| faraday                   | 1.0.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.6        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.2.2        |
| uber                      | 0.1.0        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.15.0       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.1        |
| googleauth                | 0.13.1       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.3        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-storage      | 1.28.0       |
| emoji_regex               | 3.0.0        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.365.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.2        |
| aws-sdk-core              | 3.105.0      |
| aws-sdk-kms               | 1.37.0       |
| aws-sdk-s3                | 1.79.1       |
| nanaimo                   | 0.3.0        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.18.0       |
| json                      | 2.3.1        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |
</details>

*generated on:* **2020-09-06**
</details> </pre>
</details>

**Comment by fzyzcjy** [comment]
No it does not work :(

**Comment by Collette-21** [comment]
This is not working please provide us correct solution [paycheckrecords](https://www.paycheckrecords.us/)

**Comment by fzyzcjy** [comment]
No it does not work :(

**Comment by fzyzcjy** [comment]
No it still does NOT work

**Comment by rogerluan** [comment]
Hey @fzyzcjy are you able to archive the app via Xcode IDE? What about via `xcodebuild` CLI?

That error is thrown when your app's bitcode preferences don't match with **ALL** your app's dependencies (e.g. SPM, SPM binaries, cocoapods, carthage, flutter dependencies, etc).

This explanation might help: https://stackoverflow.com/a/37612626/4075379
Also the locked issue you posted has a comment that I found useful: https://github.com/fastlane/fastlane/issues/15685#issuecomment-602925261

If you're still facing this issue, you should check whether all your dependencies have bitcode enabled. Unfortunately, the more dependencies your project has, the hairier this will get, as this information is not obvious on 3rd party's code 😬 if using cocoapods, you can use the `post_install` hook to print the dependencies' bitcode settings.

**Comment by fzyzcjy** [comment]
@rogerluan Thanks for the reply (after 3 months)! So, if I do not enable that config, how worse will it be? For example, without enabling this flag, will my app's performance be much worse, or will the size of apk be much bigger? If no big difference then I will ignore it.

**Comment by rogerluan** [comment]
@fzyzcjy I'd recommend reading about Xcode's Bitcode :) it's a way to possibly allow Appe to further optimize your app without uploading new binaries. An example of this can be e.g. if Apple introduced a new architecture 😇 
It's important to notice that it's different than slicing.

https://stackoverflow.com/questions/30722606/what-does-enable-bitcode-do-in-xcode-7

tl;dr: bitcode is not important imo, feel free to disable it.

**Comment by rogerluan** [comment]
Correction: 4 months* 😜😝 jk.

**Comment by fzyzcjy** [comment]
@rogerluan Thanks for that!
